packages/server/src/utils.ts

```diff
@@ -1,14 +1,29 @@
 import dayjs from '~/lib/dayjs';
 
-export function getActiveUntilDate(oldActiveUntil: Date, anchorDate: Date): Date {
-  const anchorDay = Math.min(dayjs(oldActiveUntil).add(1, 'month').endOf('month').date(), dayjs(anchorDate).date());
-  return dayjs(oldActiveUntil).add(1, 'month').set('date', anchorDay).toDate();
+export function getPeriodFromAnchorDate(someDateInPeriod: Date, _anchorDate: Date): { start: Date; end: Date } {
+  const startDate = Math.min(dayjs(someDateInPeriod).endOf('month').date(), dayjs(_anchorDate).date());
+  let start = dayjs(someDateInPeriod).set('date', startDate).startOf('day');
+
+  if (start.isAfter(someDateInPeriod)) {
+    // select previous period
+    let end = start.endOf('day');
+    const newStart = start.subtract(1, 'month');
+    let date = Math.max(newStart.date(), dayjs(_anchorDate).date());
+    date = Math.min(date, newStart.endOf('month').date());
+    if (date === start.date()) {
+      end = start.subtract(1, 'day').endOf('day');
+    }
+    start = newStart.set('date', date);
+    return { start: start.toDate(), end: end.toDate() };
+  }
```

Comment:
@anbraten This if block is hard to understand in my opinion. Wouldn't the following suggestion be sufficient?

```suggestion
  if (start.isAfter(someDateInPeriod)) {
    // calculate start of previous period
    const newStart = start.subtract(1, 'month');
    let date = Math.max(newStart.date(), dayjs(_anchorDate).date());
    date = Math.min(date, newStart.endOf('month').date());
    start = newStart.set('date', date);
  }
```

I also think there would have been a wrong end date in the following situation
- anchorDate `31.01.`
- someDateInPeriod `26.02.`

This would result in
- start (in line 5) `28.02.`
- end `28.02`
- newStart `28.01.`

So the end date is wrong because the if in line 13 is not true. The end date should be the `27.02.`.